docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    /// <example>
+    /// <code>
+    /// int c = Math.Add(4, 5);
+    /// if (c > 10)
+    /// {
+    ///     Console.WriteLine(c);
+    /// }
+    /// </code>
+    /// </example>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than 0.</exception>
+    /// See <see cref="Math.Add(double, double)"/> to add doubles.
+    /// <seealso cref="Math.Subtract(int, int)"/>
+    /// <seealso cref="Math.Multiply(int, int)"/>
+    /// <seealso cref="Math.Divide(int, int)"/>
+    public static int Add(int a, int b)
+    {
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+The `cref` attribute represents a reference to a type or its member that is available from the current compilation environment.
+This can be any type defined in the project or a referenced assembly.
+
+### &lt;param&gt;
+
+The `<param>` tag is used for describing the parameters a method takes. Here's an example on the double `Add` method:
+The parameter the tag describes is specified in the **required** `name` attribute.
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// <para>The main <c>Math</c> class.</para>
+/// <para>Contains all methods for performing basic math functions.</para>
+/// </summary>
+public class Math
+{
+    /// <summary>
+    /// Adds two doubles and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two doubles.
+    /// </returns>
+    /// <exception cref="System.OverflowException">Thrown when one parameter is max and the other is greater than zero.</exception>
+    /// See <see cref="Math.Add(int, int)"/> to add integers.
+    /// <param name="a">A double precision number.</param>
+    /// <param name="b">A double precision number.</param>
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+}
+```
+
+### &lt;typeparam&gt;
+
+The `<typeparam>` tag functions just like the `<param>` tag but for generic type or method declarations to describe a generic parameter.
+Since our math library doesn't have a generic method I'll just bring in `SomeClass` to demonstrate it.
```

Comment:
Your math library might include a generic "GreaterThan" method that was generic and relied on types implementation IComparable<T>:

public bool GreaterThan<T>(T left, T right) where T : IComparable<T>